### ISSUE TITLE
Remote configuration warning

### ISSUE THREAD

**Issue by alsyia** [DataDog/dd-trace-py#5212]
### Summary of problem

Hi,
We've upgraded some of our services to use dd-trace-py 1.8.0 recently and they all log the following warning:
>Agent is down or Remote Config is not enabled in the Agent
>Check your Agent version, you need an Agent running on 7.39.1 version or above.
>Check Your Remote Config environment variables on your Agent:
>DD_REMOTE_CONFIGURATION_ENABLED=true
>DD_REMOTE_CONFIGURATION_KEY=<YOUR-KEY>
>See: https://app.datadoghq.com/organization-settings/remote-config

We are running agent 7.40.1 and tried upgrading to 7.43.0 but that didn't help. We don't have the variables quoted set on the agent, but I could not find any reference to those or to remote configuration in the configuration template file, or the Helm chart variables.

I've read some blog posts and it appears remote configuration is a new future, possibly not globally available yet?

What are we suppose to do? How do we create a "remote configuration key"? 
Alternatively, would setting `DD_REMOTE_CONFIGURATION_ENABLED=false` _on pods_ remove the warning? (we would rather silence it without adding an extra environment variable on all pods, of course)

### Which version of dd-trace-py are you using?

1.8.0

### Which version of pip are you using?

Depends, but probably not relevant.

### Which libraries and their versions are you using?

Idem.

### How can we reproduce your problem?

It looks like a standard agent deployment using the Helm chart 3.3.1 (or 3.11.0) without any additional configuration related tor emote configuration + a standard dd-trace-py 1.8.0 will raise the warning.

### What is the result that you get?

The warning is a bit cryptic as the variables mentioned are not document anywhere. Besides the process of enabling remote config on the agent isn't documented either - or I missed it. Maybe because it is still in beta? But then why would dd-trace raise a warning?

Thanks! :)

**Comment by alsyia** [comment]
Hey @mabdinur,

Thanks for the quick answer! Okay, we will see what we do because adding this variable on all of our projects would take a lot of time for little benefits. Thanks for confirming it's in beta though!

**Comment by infowolfe** [comment]
> Hi @alsyia,
> 
> Thanks for opening this issue. Yes, remote configuration is currently in beta. You can disable this feature (and those warnings) by setting the following environment variable: `DD_REMOTE_CONFIGURATION_ENABLED=false`. Documentation for this feature is currently in beta and will be released soon. Let me know if this works for you.
> 
> Sorry about the trouble.

Why is this enabled by default then if it's a beta feature?

**Comment by albertdcc** [comment]
That didn't work for me.

I'm running the agent on Docker with the command below and doing a small test via a Python application. Every time my Python application runs, the alert below is issued in the console but the traces are correctly recorded in Datadog.

Command to run Datadog agent on Docker:

```
docker run -d --cgroupns host --pid host --name dd-agent --network="host" -v /var/run/docker.sock:/var/run/docker.sock:ro -v /proc/:/host/proc/:ro -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro -e DD_API_KEY="xxxxxxxxx" -e DD_ENV="dev_local" -e DD_REMOTE_CONFIGURATION_ENABLED=false gcr.io/datadoghq/agent:7
```

Output of my Python Application for tests:

```
Agent is down or Remote Config is not enabled in the Agent
Check your Agent version, you need an Agent running on 7.39.1 version or above.
Check Your Remote Config environment variables on your Agent:
DD_REMOTE_CONFIGURATION_ENABLED=true
DD_REMOTE_CONFIGURATION_KEY=<YOUR-KEY>
```

**Comment by balusarakesh** [comment]
this is happening for us too, we don't have any var like `DD_REMOTE_CONFIGURATION_ENABLED` set in our agent

**Comment by oav17** [comment]
I'm also having this error, I configured env var `DD_REMOTE_CONFIGURATION_ENABLED=false` but it's still happening. Is something else I can do?

**Comment by nodesocket** [comment]
We just started experiencing this and it’s filling our logs full of:

Mar 22 13:40:55 celery-worker-68b84f87bf-s9mmn celery-worker WARNING Agent is down or Remote Config is not enabled in the Agent
Mar 22 13:40:55 celery-worker-68b84f87bf-s9mmn celery-worker Check your Agent version, you need an Agent running on 7.39.1 version or above.
Mar 22 13:40:55 celery-worker-68b84f87bf-s9mmn celery-worker Check Your Remote Config environment variables on your Agent:
Mar 22 13:40:55 celery-worker-68b84f87bf-s9mmn celery-worker DD_REMOTE_CONFIGURATION_ENABLED=true
Mar 22 13:40:55 celery-worker-68b84f87bf-s9mmn celery-worker DD_REMOTE_CONFIGURATION_KEY=<YOUR-KEY>
Mar 22 13:40:55 celery-worker-68b84f87bf-s9mmn celery-worker See: https://app.datadoghq.com/organization-settings/remote-config

What’s the fix? Why is this enabled by default all the sudden?

**Comment by balusarakesh** [comment]
setting `DD_REMOTE_CONFIGURATION_ENABLED=false` removed those warnings from our logs

**Comment by WarpRat** [comment]
Some recent update broke the previous fix for this. When we updated from `1.18.5` to `1.20.1` these messages started showing up in our logs again.

**Comment by Tyler-2** [comment]
This is due to this line:
https://github.com/DataDog/dd-trace-py/blame/v1.20.2/ddtrace/internal/remoteconfig/worker.py#L53

And it looks like 5 hours ago they fixed it.
https://github.com/DataDog/dd-trace-py/blob/f4a675f6fc450cff9db9f16342b6d1aa5ddc6c67/ddtrace/internal/remoteconfig/worker.py#L51

But it's not yet released... I'd expect it to be in 1.20.3?

**Comment by ifmusic** [comment]
I'm also on `ddtrace==1.20.2` ; 
Agent is at: `gcr.io/datadoghq/agent:7.46.0`, 
and setting `DD_REMOTE_CONFIGURATION_ENABLED` to `false` or `true` renders the same log message.
APM and log metrics still work, but we keep getting these logs.

**Comment by khaerulumam42** [comment]
> Some recent update broke the previous fix for this. When we updated from `1.18.5` to `1.20.1` these messages started showing up in our logs again.

Does this mean it is safe to use version 1.20.0?

**Comment by jclerman** [comment]
> This is due to this line: https://github.com/DataDog/dd-trace-py/blame/v1.20.2/ddtrace/internal/remoteconfig/worker.py#L53
> 
> And it looks like 5 hours ago they fixed it.
> 
> https://github.com/DataDog/dd-trace-py/blob/f4a675f6fc450cff9db9f16342b6d1aa5ddc6c67/ddtrace/internal/remoteconfig/worker.py#L51
> 
> But it's not yet released... I'd expect it to be in 1.20.3?

It was not backported to 1.20.3.  This probably needs to be filed as a new ticket to get attention?

**Comment by Tyler-2** [comment]
Yeah, it's not in 1.20.3 Come ooooon. They're making money off of the added log volume too.

**Comment by StarLightPL** [comment]
Setting DD_REMOTE_CONFIGURATION_ENABLED false worked as a workaround for me.

**Comment by chamini2** [comment]
I was experiencing this in 1.18.3 and moved to 1.18.1 while it gets fixed, they stopped appearing.

**Comment by lightswitch05** [comment]
Checking in that this is still an issue in [1.20.5](https://github.com/DataDog/dd-trace-py/releases/tag/v1.20.5). Hopefully we can get this back-ported to 1.20.x?

**Comment by juanjux** [comment]
Yes, there was a regression and was fixed, however is seems the backport to 1.20 still has not merged, we'll take care of it and release a new minor version with the fix.

**Comment by juanjux** [comment]
We will release 1.20.9 in the next few hours which should fix (again) this. Thanks for your patience.
